src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,22 @@
+import Popup from './Popup.js';
+
+export default class PopupWithConfirmation extends Popup {
+    constructor(popupSelector) {
+        super({ popupSelector });
+        this._form = this._popupElement.querySelector('.modal__form');
+    }
+
+    // Set action when the user clicks on the confirmation button
+    setAction(action) {
+        this._handleFormSubmit = action;
+    }
+
+    // Set event listener on the form element
+    setEventListeners() {
+        this._form.addEventListener('submit', () => { 
```

Comment:
You should also remove the listener when closing the popup. Don't forget, that `addEventListener` and `removeEventListener` accept the same arguments comparing by reference

Reply:
That's done! Thank you for the reminder. :)